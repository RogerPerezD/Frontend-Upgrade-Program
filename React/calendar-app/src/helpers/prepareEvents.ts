import moment from 'moment';
import { Event } from '../reducers/calendarReducer';

export const prepareEvents = ( events: Event []) =>{
    // console.log(events);

    return events.map( (event) => {
        return {
            ...event,
            end: moment( event.end ).toDate(),
            start: moment( event.start ).toDate(),
        }
    })
}