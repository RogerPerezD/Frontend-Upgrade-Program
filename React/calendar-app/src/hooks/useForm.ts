import { FormEvent, useState } from "react";


export const useForm = <T>( initialState: T ): [ T, (e: FormEvent<HTMLInputElement>)=>void, ()=>void ] => {
    const [values, setValues] = useState<T>(initialState);

    const handleInput = ( { currentTarget }: FormEvent<HTMLInputElement> ): void =>{
        setValues({
            ...values,
            [ currentTarget.name ]: currentTarget.value
        });
    }

    const reset  = () =>{
        setValues( initialState );
    }

    return [ values, handleInput, reset ];
};