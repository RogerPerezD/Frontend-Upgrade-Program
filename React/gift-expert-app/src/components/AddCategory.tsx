import React,{ useState } from 'react';

interface Props{
    setCategories: React.Dispatch<React.SetStateAction<string[]>>;
}

const AddCategory: React.FC<Props> = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleChangeInput = ( event: React.FormEvent<HTMLInputElement>) =>{
        setInputValue( event.currentTarget.value );
    }

    const handleSubmit = async( event: React.FormEvent) =>{
        event.preventDefault();
        console.log('Sumit launch', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories( (categories: string[]) => { 
                return [inputValue,...categories];
            });
        }
        setInputValue('');
    }


    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue } 
                onChange={ handleChangeInput }
            />
            <button type='submit'>Agregar</button>
        </form>
    );
}

export default AddCategory;
