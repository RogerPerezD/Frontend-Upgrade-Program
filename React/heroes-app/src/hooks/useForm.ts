import { FormEvent, useState } from "react";

type FieldsForm = {
    searchText: string
}

export const useForm = ( initialState: FieldsForm ): [FieldsForm, (e: FormEvent<HTMLInputElement>)=>void, ()=>void ] => {
    const [values, setValues] = useState<FieldsForm>(initialState);

    const handleInput = ( { currentTarget }: FormEvent<HTMLInputElement> ): void =>{
        setValues({
            ...values,
            [ currentTarget.name ]: currentTarget.value
        });
    }

    const reset  = () =>{
        setValues( initialState );
    }

    return [ values, handleInput, reset ];
};
