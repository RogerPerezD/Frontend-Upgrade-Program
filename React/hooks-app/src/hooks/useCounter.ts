import { useState } from "react";



const useCounter = ( initialState: number = 0) => {
  
    const [state, setState] = useState(initialState);

    const increment = ( factor: number = 1) =>{
        setState( state + factor);
    }

    const decrement = ( factor: number = 1) =>{
        setState( state - factor);
    }

    const reset = () =>{
        setState( initialState );
    }

    return {
        state,
        increment,
        decrement,
        reset
    };
};

export default useCounter;
