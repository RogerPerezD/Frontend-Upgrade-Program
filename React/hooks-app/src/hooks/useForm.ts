import { FormEvent, useState } from "react";
import { FieldsForm } from "../components/02-useEffect/FormWithCustomHook";


export const useForm = ( initialState: FieldsForm ): [FieldsForm, (e: FormEvent<HTMLInputElement>)=>void, ()=>void ] => {
    const [values, setValues] = useState<FieldsForm>(initialState);

    const handleInput = ( { currentTarget }: FormEvent<HTMLInputElement> ): void =>{
        setValues({
            ...values,
            [ currentTarget.name ]: currentTarget.value
        });
    }

    const reset  = () =>{
        setValues( initialState );
    }

    return [ values, handleInput, reset ];
};
