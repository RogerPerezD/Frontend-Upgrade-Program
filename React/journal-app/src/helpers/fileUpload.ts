
export const fileUpload = async (file: File | Buffer) =>{

    const cloudUrl = 'https://api.cloudinary.com/v1_1/rogerchdk/upload';

    const formData = new FormData();
    formData.append('upload_preset', 'react-journal');
    formData.append('file', file as File);

    try {
        const resp = await fetch( cloudUrl, {
            method: 'POST',
            body: formData
        });

        if (resp.ok) {
            const cloudResp = await resp.json();
            return cloudResp.secure_url;
        }else{
            return null;
        }
    } catch (error) {
        throw error;
    }
}