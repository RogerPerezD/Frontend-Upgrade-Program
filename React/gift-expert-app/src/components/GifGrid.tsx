
// import React, { useEffect, useState } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

interface Props{
    category: string
}

export const GifGrid: React.FC<Props> = ({ category }) => { 
    
    const {data: images, loading} = useFetchGifs( {category} );
    return (
        <>
        <h3 className="animate__animated animate__bounce">{ category }</h3>
        { loading && <p className="animate__animated animate__flash">Loading</p>}
        <div className='card-grid'>
            <ol>
                {
                    images?.map( (elemet: {
                        id: string,
                        title: string,
                        url: string
                      }) =>{
                        return  <GifGridItem key={ elemet.id} { ...elemet}/>
                    })
                }
            </ol>
        </div>
        </>
    );
}
