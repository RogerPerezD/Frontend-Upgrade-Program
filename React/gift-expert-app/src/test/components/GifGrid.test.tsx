import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import { Gif } from '../../helpers/getGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Test in <GifGrid/>', () => {

    const mockUseFetchGifs = useFetchGifs as jest.MockedFunction<typeof useFetchGifs>;
    
    test('should show the component succesfully', () => {
        
        
        mockUseFetchGifs.mockReturnValue({
            data: null,
            loading: true
        });
        const wrapper = shallow( <GifGrid category='naruto'/>);

        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the items when the images are loaded whit useFetchGifs', () => {
        
        const gifs: Gif = [{
            id: 'asd',
            title: 'Dreagon ball',
            url: 'https://localhost/cosa.jpg'
        }];
        
        mockUseFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category='naruto'/>);

        expect( wrapper.find('p').exists()).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });

    


});