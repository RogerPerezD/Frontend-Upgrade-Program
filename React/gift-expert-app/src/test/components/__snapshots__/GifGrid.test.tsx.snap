// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGrid/> should show the component succesfully 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
